src/bun.js/test/jest.zig

```diff
@@ -430,6 +493,213 @@ pub const TestRunner = struct {
+                values = ValuesHashMap.init(default_allocator);
+
+                var dir_path = runner.?.files.get(current_file_id).source.path.name.dirWithTrailingSlash();
+                if (!initialized) snapshot_dir = try getSnapshotDir(dir_path);
+                snapshot_file = try getSnapshotFile(current_file_id);
+
+                if (update_snapshots) {
+                    file_buf = try std.ArrayList(u8).initCapacity(default_allocator, file_header.len);
+                    file_buf.appendSliceAssumeCapacity(file_header);
+                } else {
+                    const file_length = try snapshot_file.getEndPos();
+                    const bytes = try snapshot_file.readToEndAlloc(default_allocator, file_length);
+                    file_buf = std.ArrayList(u8).fromOwnedSlice(default_allocator, bytes);
+
+                    try parseFile();
+                }
+
+                initialized = initialized or true;
+            }
+
+            snapshot_count += 1;
+
+            var count_entry = try counts.getOrPut(snapshot_name);
+            const counter = brk: {
+                if (count_entry.found_existing) {
+                    default_allocator.free(snapshot_name);
+                    count_entry.value_ptr.* += 1;
+                    break :brk count_entry.value_ptr.*;
+                }
+                count_entry.value_ptr.* = 1;
+                break :brk count_entry.value_ptr.*;
+            };
+
+            const name = count_entry.key_ptr.*;
+
+            var counter_string_buf = [_]u8{0} ** 32;
+            var counter_string = try std.fmt.bufPrint(&counter_string_buf, "{d}", .{counter});
+
+            var name_with_counter = try default_allocator.alloc(u8, name.len + 1 + counter_string.len);
+            defer default_allocator.free(name_with_counter);
+            bun.copy(u8, name_with_counter[0..name.len], name);
+            name_with_counter[name.len] = ' ';
+            bun.copy(u8, name_with_counter[name.len + 1 ..], counter_string);
+
+            const name_hash = std.hash.Wyhash.hash(0, name_with_counter);
+            if (values.get(name_hash)) |expected| {
+                return expected;
+            }
+
+            // doesn't exist. append to file bytes and add to hashmap.
+            var pretty_value = try MutableString.init(default_allocator, 0);
+            try value.jestSnapshotPrettyFormat(&pretty_value, globalObject);
+
+            try file_buf.appendSlice("\nexports[`");
+            try file_buf.appendSlice(name_with_counter);
+            try file_buf.appendSlice("`] = `");
+            try file_buf.appendSlice(pretty_value.list.items);
+            try file_buf.appendSlice("`;\n");
+            try values.put(name_hash, pretty_value.toOwnedSlice());
+            return null;
+        }
+
+        pub fn parseFile() !void {
+            var i: usize = 0;
+            outer: while (i < file_buf.items.len) {
+                const key_start: usize = brk: {
```

Comment:
while the effort to be simple/fast is good

i worry this is going to be too unreliable

wdyt about using the JS parser here?

Reply:
switched to JS parser